XenModel/Network/TaskPoller.cs

```diff
@@ -98,55 +98,51 @@ public void PollToCompletion()
 
         private void poll()
         {
-            try
-            {
-                XenAPI.Task task = GetTask();
-                _action.Tick((int)(task.progress * _scale + _lo),
-                             task.Description() == "" ? _action.Description : task.Description());
-                switch (task.status)
-                {
-                    case XenAPI.task_status_type.failure:
-                        log.Warn("Action failed due to API failure:\n" + Environment.StackTrace);
-                        throw new XenAPI.Failure(new List<string>(task.error_info));
+            Task task = GetTask();

```

Comment:
Is it possible that `GetTask()` throws the HANDLE_INVALID failure instead of returning a task with `task_status_type.failure`?